base/gmp.jl

```diff
@@ -521,8 +521,11 @@ hex(n::BigInt) = base(16, n)
 for f in (:bin, :oct, :dec, :hex)
     @eval function ($f)(n::BigInt, pad::Int)
         b = IOBuffer()
-        res = ($f)(n)
+        res = ($f)(abs(n))
```

Comment:
> One way to do it would be to temporarily construct abs(n) while sharing the same data pointer as n, but that may be a bit too tricky.
couldn't we expect this tricks from the compiler? after all n is immuatable and abs(n) is local inside an 
expression. 

>Using GMP printing functions directly is probably the better approach.
which printing functions of GMP do you refer to? I only found get_...z_str, which does not support padding in the docu of current version 6.1.1.